query/excluder.go

```diff
@@ -0,0 +1,94 @@
+package query
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+)
+
+//Excluder allows us to support different methods of excluding in the future
+type Excluder interface {
+	ShouldExclude(path string) bool
+}
+
+//RegexpExclude uses regular expressions to tell if a file/path should be excluded
+type RegexpExclude struct {
+	recursive  bool
+	exclusions []string
+	regex      string
```

Comment:
I think it'd be more appropriate if this was of type [`*regexp.Regexp`](https://golang.org/pkg/regexp/#Regexp).